### PR TITLE
Fix issue with spurious timing test failures

### PR DESCRIPTION
**Pull request by JasonGross**
When none of the numbers get over 100, the width of the table was
different.

See https://github.com/coq/coq/pull/6736#issuecomment-365386802

### CODE REVIEW

**Review by ejgallego — COMMENTED** [review]
I'm afraid I cannot understand how the sed system works, but I trust Jason here and this should fix a real issue we've seen in the CI.

File: test-suite/coq-makefile/timing/run.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ echo
 
 for ext in "" .desired; do
     for file in A.v.timing.diff; do
-        cat ${file}${ext} | sed s"/${INFINITY}/${INFINITY_REPLACEMENT}/g" | sed s":|\s*N/A\s*$:| ${INFINITY_REPLACEMENT}:g" | sed s'/[0-9]*\.[0-9]*//g' | sed s'/0//g' | sed s'/  */ /g' | sed s'/ *$//g' | sed s'/+/-/g' | sort > ${file}${ext}.processed
+        cat ${file}${ext} | sed s"/${INFINITY}/${INFINITY_REPLACEMENT}/g" | sed s":|\s*N/A\s*$:| ${INFINITY_REPLACEMENT}:g" | sed s'/[0-9]*\.[0-9]*//g' | sed s'/0//g' | sed s'/^-*$/------/g' | sed s'/  */ /g' | sed s'/ *$//g' | sed s'/+/-/g' | sort > ${file}${ext}.processed
```
Comment (SkySkimmer) [inline test-suite/coq-makefile/timing/run.sh]: This is pretty unreadable. How about a cleanup step using an array to gather the sed expressions:
```bash
TOSED_IN=(
    s"/${INFINITY}/${INFINITY_REPLACEMENT}/g"
    s'/[0-9]//g'
    s'/ *$//g'
    s":|\s*N/A\s*$:| ${INFINITY_REPLACEMENT}:g"
    s'/^-*$/------/g'
    s'/  */ /g'
    s'/\(Total.*\)-\(.*\)-/\1+\2+/g'
)

# we need to add -e before each
TOSED=()
for s in "${TOSED_IN[@]}"; do TOSED+=("-e" "$s"); done

for ext in "" .desired; do
    for file in time-of-build-before.log time-of-build-after.log time-of-build-both.log; do
        cat ${file}${ext} | grep -v 'warning: undefined variable' | sed "${TOSED[@]}" > ${file}${ext}.processed
```
(this is for the 1st sed block at line 43, not the one this comment is attached to)
Each expression (line) could have a comment explaining why it's necessary.

Also there are 2 similar looking sed blocks, should they be the same? In which case this PR is missing the first one.

Comment (SkySkimmer) [reply test-suite/coq-makefile/timing/run.sh]: Or even using `sed -f`.

**Comment by maximedenes** [comment]
@JasonGross What do you think of @SkySkimmer 's suggestion?

File: test-suite/coq-makefile/timing/run.sh (hunk 1 above)
Comment (JasonGross) [reply test-suite/coq-makefile/timing/run.sh]: There should not be enough variation that the first block needs this, but I've added it anyway.  I now use bash arrays and have added comments

**Comment by JasonGross** [comment]
@maximedenes I've implemented it (but haven't tested, so definitely wait for travis)

File: test-suite/coq-makefile/timing/run.sh [hunk 2]
```diff
@@ -38,9 +38,29 @@ make print-pretty-timed-diff TIMING_SORT_BY=diff || exit $?
...
+TO_SED_IN_PER_LINE=(
+    s'/0//g' # unclear whether this is actually needed above and beyond s'/[0-9]*\.[0-9]*//g'; it's been here from the start
+    s'/  */ /g' # Sometimes 0 will show up as 0m00.s, sometimes it'll end up being more like 0m00.001s; we must strip out the spaces that result from left-aligning numbers of different widths based on how many digits Coq's [-time] gives
+    s'/+/-/g' # some code lines don't really change, but this can show up as either -0m00.01s or +0m00.01s, so we need to normalize the signs
+    )
+
 for ext in "" .desired; do
     for file in time-of-build-before.log time-of-build-after.log time-of-build-both.log; do
-        cat ${file}${ext} | grep -v 'warning: undefined variable' | sed s"/${INFINITY}/${INFINITY_REPLACEMENT}/g" | sed s'/[0-9]//g' | sed s'/ *$//g' | sed s":|\s*N/A\s*$:| ${INFINITY_REPLACEMENT}:g" | sed s'/^-*$/------/g' | sed s'/  */ /g' | sed s'/\(Total.*\)-\(.*\)-/\1+\2+/g' > ${file}${ext}.processed
+        cat ${file}${ext} | grep -v 'warning: undefined variable' | sed "${TO_SED_IN_BOTH[@]}" | sed "${TO_SED_IN_PER_FILE[@]" > ${file}${ext}.processed
```
Comment (SkySkimmer) [inline test-suite/coq-makefile/timing/run.sh]: This won't work, sed needs the -e flag when you send multiple commands.
ie
```bash
$ echo ab | sed 's/a/c/' 's/b/d/'
sed: can't read s/b/d/: No such file or directory
```

Comment (JasonGross) [reply test-suite/coq-makefile/timing/run.sh]: Ah, okay.  You're the one who claimed `sed "${TOSED[@]}"` would work :-P

Comment (JasonGross) [reply test-suite/coq-makefile/timing/run.sh]: Updated to use `sed -e`

Comment (SkySkimmer) [reply test-suite/coq-makefile/timing/run.sh]: >Ah, okay. You're the one who claimed sed "${TOSED[@]}" would work :-P

I didn't, see https://github.com/coq/coq/pull/6756#discussion_r168028764
>`# we need to add -e before each`

File: test-suite/coq-makefile/timing/run.sh [hunk 3]
```diff
@@ -38,9 +38,29 @@ make print-pretty-timed-diff TIMING_SORT_BY=diff || exit $?
...
+TO_SED_IN_PER_LINE=(
+    s'/0//g' # unclear whether this is actually needed above and beyond s'/[0-9]*\.[0-9]*//g'; it's been here from the start
+    s'/  */ /g' # Sometimes 0 will show up as 0m00.s, sometimes it'll end up being more like 0m00.001s; we must strip out the spaces that result from left-aligning numbers of different widths based on how many digits Coq's [-time] gives
+    s'/+/-/g' # some code lines don't really change, but this can show up as either -0m00.01s or +0m00.01s, so we need to normalize the signs
+    )
+
 for ext in "" .desired; do
     for file in time-of-build-before.log time-of-build-after.log time-of-build-both.log; do
-        cat ${file}${ext} | grep -v 'warning: undefined variable' | sed s"/${INFINITY}/${INFINITY_REPLACEMENT}/g" | sed s'/[0-9]//g' | sed s'/ *$//g' | sed s":|\s*N/A\s*$:| ${INFINITY_REPLACEMENT}:g" | sed s'/^-*$/------/g' | sed s'/  */ /g' | sed s'/\(Total.*\)-\(.*\)-/\1+\2+/g' > ${file}${ext}.processed
+        cat ${file}${ext} | grep -v 'warning: undefined variable' | sed -e "${TO_SED_IN_BOTH[@]}" | sed -e "${TO_SED_IN_PER_FILE[@]}" > ${file}${ext}.processed
```
Comment (SkySkimmer) [inline test-suite/coq-makefile/timing/run.sh]: This still doesn't work, you need one -e for each script.
You can put them in the array (`TO_SED=(-e s/foo/bar/ -e /x/y/ ...)`) or modify the array as in my previous comment. The first is probably clearer.

Comment (JasonGross) [reply test-suite/coq-makefile/timing/run.sh]: Thanks.  Updated.  I'm now testing it locally.

**Review by SkySkimmer — APPROVED** [review]
If CI doesn't explode this should be ready.

**Comment by JasonGross** [comment]
`make test-suite` works for me, locally

**Pull request merged**